packages/nextjs-kit/src/types/index.ts

```diff
@@ -21,3 +21,13 @@ export type FooterMenuItem = LinkProps & {
  */
 export const hasParent = (item: FooterMenuItem): item is FooterMenuItem =>
   item.parentId ? true : item.parent ? true : false;
+
+export const isNumber = (item: number | null | undefined): item is number =>
+  typeof item === 'number';
+export interface PaginatorObj {
+  id: string;
+  title: string;
+  body: {
+    value: string;
+  };
+}
```

Comment:
This type is fine for our default RederData component, but if the user passes in their own component with data that does not fit this shape, I'm worried this will cause problems.

Can we make the type for the data more dynamic here? Possibly use a generic so that a user can pass in a type of data that we don't necessarily know about.